roles/cfg_openwrt/templates/uplink_gateway/nftables.d/20-gre-maxseg-size.nft.j2

```diff
@@ -0,0 +1,14 @@
+{% if openwrt_version == 'snapshot' %}
+{% set TCP_HEADER_SIZE = 20 %}
+{% set IPV4_HEADER_SIZE = 20 %}
+{% set IPV6_HEADER_SIZE = 40 %}
+{% set GRE_HEADER_SIZE = 24 %}
+{% set MTU = 1500 %}
+{% set ipv4_mss = MTU - GRE_HEADER_SIZE - TCP_HEADER_SIZE - IPV4_HEADER_SIZE %}
+{% set ipv6_mss = MTU - GRE_HEADER_SIZE - TCP_HEADER_SIZE - IPV6_HEADER_SIZE %}
+chain gre_maxseg {
+    type filter hook forward priority -2; policy accept;
+	oifname "gre*" tcp flags syn tcp option maxseg size set meta nfproto map { ipv4 : {{ ipv4_mss }}, ipv6 : {{ ipv6_mss }} }
```

Comment:
why do you adjust ipv4 mss? we dont route over gre. sure it seems more complete but please write a comment its never being used

Reply:
I wrote a comment that is only visible in full file overview